src/loaders/abstract_loader.py

```diff
@@ -44,12 +44,14 @@ class AbstractVideoLoader(metaclass=ABCMeta):
     """
 
     def __init__(self, video_metadata: VideoMetaInfo, batch_size=1,
-                 skip_frames=0, offset=None, limit=None):
+                 skip_frames=0, offset=None, limit=None, threshold=0, distance_metric='absolute_difference'):
```

Comment:
This is an abstract class and contains those attributes which are common to all loaders. 

threshold and distance_metric look very specific to your use-case. You can create your own loader which inherits the AbstractVideoLoader and add there attributes to that class.